pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -55,10 +48,17 @@ const (
 	// leaking a blob that is no longer tagged in given repository.
 	BlobRepositoryCacheTTLEnvVar = "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_BLOBREPOSITORYCACHETTL"
 
+	// ManifestStorageEnvVar defines the storage type for manifests.
+	ManifestStorageEnvVar = "REGISTRY_MIDDLEWARE_MANIFEST_STORAGE"
+
 	// Default values
 
 	defaultDigestToRepositoryCacheSize = 2048
 	defaultBlobRepositoryCacheTTL      = time.Minute * 10
+
+	// Possible storages for manifests.
+	manifestStorageEtcd     = "etcd"
```

Comment:
@smarterclayton As an alternative can be faster storage (e.g. mongodb). For another storage need to implement the `distribution.ManifestService` interface. I didn't want to hardcode the docker-registry storage and leave the possibility to change it in the future.